### DISCUSSION TITLE
Fraud prevention - Signature for purchases

### DISCUSSION THREAD

**Discussion by jxs714** [pretix/pretix#3939] [Q&A] [answered]
Hi Raphael,

Previously, on 2023 versions, I tried to use the signature question plugin and it did not function properly. It would not show the signature in the backed when I set it up based on the instructions. I also could not download the signature file when the ticket showed it captured it. When I tried it in 2024 version, pretix did not perform properly. Whenever I tried to create a new ticket product, I would get an internal server error. When I removed the plugin, things went back to working order. Not sure if that page was the only page that showed an internal server error.

My main question for this topic was wanting to know if you guys would be updating or adding this feature natively, or any new features to take steps against chargebacks. I previously used a another ticketing platform that connected with stripe and a customer decided to do a chargeback (even though they agreed to our terms with text as signature and attended our event) and won the case because Stripe and their bank accepted their false fraud claim. They do consider other factors, one being some sort of proof with a hand written signature to the terms and conditions. 

It was very frustrating and now getting into these possible matters for the future, I grow a concern of having to deal with this issue in the future.. Do you have any advice or upcoming features / updates to the signature plugin or anything that is being considered in new updates that combats false chargeback claims?

**Comment by raphaelm** [comment]
How does the signature help against fraud if you have nothing to compare it with?

**Reply by jxs714** [reply]
Thanks for the response. There are several scenarios:

- The purchaser and purchase is legit but a false fraud claim was filed because you denied their refund request for a non-refundable item prior to the event, or they attended your event or received product and is intentionally committing fraud
- The purchase is actual fraud (ie: stolen card) that successfully captured due to matching zip, address, cvv, etc.

You cant control the actual fraud purchases, you can only try to prevent them through Stripes fraud protection algorithm / filter backend. However, legit purchases where the purchaser willfully tried to commit theft of services / product and who provided a legit hand written signature during checkout gives some concrete proof (vs typing a full name that anyone can do) for the bank to review. The bank will be able to review past purchases and signature history to compare hand signatures with the one provided during checkout. Although they can also fake the initial purchase with a scrabble signature (if the purchase was pre-meditated fraud from the beginning), other factors in combination can also be used as proof such as: IP address location during purchase with timestamps, etc. Overall nothing is promised from the digital side, but its a good starting point. The next step for prevention would be in person ID checking (that you can inform them of during checkout) to match the names of each ticket purchase and keep some kind of manual log of ID # etc. This can done outside of pretix unless you guys also consider logging/scanning IDs to attach to their profile when scanning qr code.

Hope this helps

**Comment by jxs714** [comment]
Hi @raphaelm 

After re-installing the Pretix Signature Question plugin, I was able to get the signature plugin to work. However, there seems to be a security concern when using the plugin as the code "unsafe-inline" as the browser developer tools states.

Using "unsafe-inline" in the CSP makes XSS attacks possible. Shouldn't this be a concern when using the plugin? When not using any CSP headers, here is the error in developer tools:

Content Security Policy blocks inline execution of scripts and stylesheets
The Content Security Policy (CSP) prevents cross-site scripting attacks by blocking inline execution of scripts and style sheets.
To solve this, move all inline scripts (e.g. onclick=[JS code]) and styles into external files.
⚠️ Allowing inline execution comes at the risk of script injection via injection of HTML script elements. If you absolutely must, you can allow inline script and styles by:
adding unsafe-inline as a source to the CSP header
adding the hash or nonce of the inline script to your CSP header.
2 directives
Directive	Element	Source location	Status
style-src-attr		output.238620df79f2.js:1	blocked
style-src-attr		output.238620df79f2.js:1	blocked

Using unsafe-inline is never recommended. Is there a specific setup that should be configured to make it operate in a secured way? Or should I remove all CSP and use as is while ignoring the first error? The plugin works fine with the setup, but I just want to be sure Im not missing anything as far as security when using the plugin

**Reply by raphaelm** [reply, marked as answer]
Sorry, I don't provide support/maintenance for that plugin. If everything works, the error can likely be ignored.
